pkg/query-service/app/querier/querier.go

```diff
@@ -149,7 +149,8 @@ func (q *querier) execPromQuery(ctx context.Context, params *model.QueryRangePar
 //
 // The [End - fluxInterval, End] is always added to the list of misses, because
 // the data might still be in flux and not yet available in the database.
-func findMissingTimeRanges(start, end, step int64, seriesList []*v3.Series, fluxInterval time.Duration) (misses []missInterval) {
+func findMissingTimeRanges(start, end, step int64, seriesList []*v3.Series, fluxInterval time.Duration) (misses []missInterval, replaceCacheData bool) {
```

Comment:
LGTM. Add a todo item to remove the `replaceCacheData` and instead use the list.

Reply:
> Add a todo item to remove the replaceCacheData and instead use the list.

Didn't get you on this

Reply:
I meant replacing is not a good idea and it should be removed eventually with better logic so please leave a TODO item in the code comments.